hw/xbox/xid-sdl.c

```diff
@@ -194,36 +195,42 @@ static const XIDDesc desc_xid_xbox_gamepad = {
 
 #define BUTTON_MASK(button) (1 << ((button) - GAMEPAD_DPAD_UP))
 
-static void update_output(USBXIDState *s) {
-#ifdef FORCE_FEEDBACK
-    if (s->sdl_haptic == NULL) { return; }
-
-    SDL_HapticLeftRight effect = {
-        .type = SDL_HAPTIC_LEFTRIGHT,
-        .length = SDL_HAPTIC_INFINITY,
-        /* FIXME: Might be left/right inverted */
-        .large_magnitude = s->out_state.right_actuator_strength,
-        .small_magnitude = s->out_state.left_actuator_strength
-    };
+static SDL_HapticEffect haptic_effect_init(USBXIDState *s, float large_magnitude_percent, float small_magnitude_percent) {
+    if (large_magnitude_percent > 1) {
```

Comment:
Should I even clamp these? and if so, should I clamp numbers below `0` too?

Reply:
If so, clamp in both directions, probably using MIN / MAX functions or something (I'm sure QEMU has something).

However, I wouldn't bother with it; maybe `assert((strength >= 0.0) && (strength <= 1.0))`?